### PR TITLE
feat(autok3s): support airgap API function

### PR DESCRIPTION
**Pull request by orangedeng**
Related issue: https://github.com/cnrancher/autok3s/issues/448
The airgap API function is under /v1/pacakges and supports:
- create
- update
- remove
- list/get
- import
- export
- update install script

### CODE REVIEW

File: pkg/server/store/pkg/store.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+		return types.APIObject{}, apierror.WrapFieldAPIError(err, validation.InvalidBodyContent, "archs", "arch(s) are invalid")
+	}
+	rtn.State = common.PackageOutOfSync
+	if err := common.DefaultDB.SavePackage(rtn); err != nil {
+		return types.APIObject{}, err
+	}
+	apiObj := common.GetAPIObject(rtn)
+
+	go downloadAndUpdatepackage(rtn)
+	return *apiObj, nil
```
Comment (Jason-ZW) [inline pkg/server/store/pkg/store.go:74]: Here it is best to unify with the `Update` method

File: pkg/server/store/pkg/store.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+}
+
+func (e *Store) Update(apiOp *types.APIRequest, schema *types.APISchema, data types.APIObject, id string) (types.APIObject, error) {
+	currents, err := common.DefaultDB.ListPackages(&id)
+	if err != nil && err != gorm.ErrRecordNotFound {
+		return types.APIObject{}, apierror.WrapAPIError(err, validation.ServerError, "failed to get package")
+	}
+	if err == gorm.ErrRecordNotFound {
+		return types.APIObject{}, apierror.NewAPIError(validation.NotFound, "package not found")
+	}
```
Comment (Jason-ZW) [inline pkg/server/store/pkg/store.go:84]:
```go
if err != nil {
  if err == gorm.ErrRecordNotFound {
     return xxxx
  }

  return xxxx
}
```

File: pkg/server/store/pkg/schema.go [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+	buff := bytes.NewBuffer([]byte{})
+	if err := settings.GetScriptFromSource(buff); err != nil {
+		apiContext.WriteError(apierror.WrapAPIError(err, validation.ServerError, "failed to get script from source"))
+		return
+	}
+	if err := settings.InstallScript.Set(buff.String()); err != nil {
+		apiContext.WriteError(apierror.WrapAPIError(err, validation.ServerError, "failed to update stored install script"))
+		return
+	}
+	w.WriteHeader(200)
```
Comment (Jason-ZW) [inline pkg/server/store/pkg/schema.go:128]: change to `http.StatusOK`?

**Review by Jason-ZW — CHANGES_REQUESTED** [review]
Others LGTM

**Comment by orangedeng** [comment]
Updated

**Review by Jason-ZW — APPROVED** [review]

**Pull request merged**